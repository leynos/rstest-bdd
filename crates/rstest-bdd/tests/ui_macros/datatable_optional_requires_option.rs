//! Compile-fail fixture: `#[datatable(optional)]` requires `Option<T>` fields.

use rstest_bdd_macros::DataTableRow;

#[derive(Debug, DataTableRow)]
struct InvalidOptionalField {
    #[datatable(optional)]
    value: String,
}

fn main() {}
