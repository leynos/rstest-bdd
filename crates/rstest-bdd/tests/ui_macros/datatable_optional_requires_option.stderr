error: #[datatable(optional)] requires an Option<T> field
 --> tests/ui_macros/datatable_optional_requires_option.rs:7:5
  |
7 | /     #[datatable(optional)]
8 | |     value: String,
  | |_________________^
